document/html/index.bs

```diff
@@ -0,0 +1,389 @@
+        text: ExportedFunction; url: #exportedfunction
+    type: dfn
+        text: compile a WebAssembly module; url: #compile-a-webassembly-module
+        text: instantiate a WebAssembly module; url: #instantiate-a-webassembly-module
+        text: instantiate; url: #dom-webassembly-instantiate
+        text: asynchronously compile a webassembly module; url: #asynchronously-compile-a-webassembly-module
+        text: instantiate a promise of a module; url: #instantiate-a-promise-of-a-module
+</pre>
+
+<pre class='link-defaults'>
+spec:infra; type:dfn; text:list
+spec:ecma-262; type:exception; for:ECMAScript; text:Error
+spec:infra; type:dfn; for:set; text:append
+</pre>
+
+This document builds off of the WebAssembly specification [[WEBASSEMBLY]] and the WebAssembly JavaScript embedding [[WASMJS]].
+
+<h2 id="streaming-modules">Streaming module APIs</h2>
+
+<pre class="idl">
+partial namespace WebAssembly {
+  Promise&lt;Module> compileStreaming(Promise&lt;Response> source);
+  Promise&lt;WebAssemblyInstantiatedSource> instantiateStreaming(
+      Promise&lt;Response> source, optional InstanceImportsMap importObject);
+};
+</pre>
+
+<h3 id="streaming-compliaton">Streaming module compilation</h3>
+
+Note: This method was not included in the initial Wasm specification, so developers may need to provide a code path for the case where it is not present.
+
+Note: This function accepts a {{Response}} object, or a
+promise for one, and compiles the resulting bytes of the response. This compilation can be performed
+in the background and in a streaming manner. If the {{Response}} is not [=CORS-same-origin=],
+does not represent an [=ok status=], or does not match the
+`application/wasm` MIME type, the returned promise will be rejected with a [=TypeError=]; if
+compilation fails, the returned promise will be rejected with a {{CompileError}}.
+
+<div algorithm>
+The <dfn method for="WebAssembly">compileStreaming(source)</dfn> method, when invoked, returns the result of [=compile a potential WebAssembly response|compiling a potential WebAssembly response=] with |source|.
+</div>
+
+<h3 id="streaming-instantiation">`WebAssembly.instantiateStreaming`</h3>
+
+Note: This method was not included in the initial Wasm specification, so developers may need to provide a code path for the case where it is not present.
+
+<div algorithm>
+The <dfn method for="WebAssembly">instantiateStreaming(source, importObject)</dfn> method, when invoked, performs the following steps:
+
+    1. Let |promiseOfModule| be the result of [=compile a potential WebAssembly response|compiling a potential WebAssembly response=] with |source|.
+    1. Return the result of [=instantiate a promise of a module|instantiating the promise of a module=] |promiseOfModule| with imports |importObject|.
+</div>
+
+<h3 id="process-response">Compile a potential WebAssembly response</h3>
+
+Note: This algorithm accepts a {{Response}} object, or a
```

Comment:
This is pretty duplicative. I'd save such descriptive notes for the web-developer-facing docs.

Reply:
You wrote these notes! Anyway I'm fine to delete them.